packages/styles/src/useStyles.ts

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useContext, useMemo } from 'react';
+import { RendererContext, ThemeContext } from 'react-fela';
+import { G } from '@mobily/ts-belt';
+import { darken, lighten, transparentize } from 'polished';
+import { breakpointMediaQueries } from './breakpoints';
+import {
+    NativeStyle,
+    NativeStyleObject,
+    NativeStyleOrStylesParam,
+    NativeStyleUtils,
+    Style,
+    StyleOrStylesParam,
+    StyleUtils,
+} from './types';
+import { Theme, NativeTheme } from '@trezor/theme';
+import { multiply, getValueAndUnit, sum } from './utils';
+import { processStyles, processNativeStyles } from './processStyles';
+import { mediaQueries } from './mediaQueries';
+import { useDirection } from './useDirection';
+
+export const useStyles = () => {
+    const renderer = useContext(RendererContext);
+
+    if (!renderer) {
+        throw new Error('The `useStyles()` hook can only be used inside a `StylesProvider`.');
+    }
+
+    const theme = useContext(ThemeContext) as Theme;
+    const nativeTheme = useContext(ThemeContext) as NativeTheme;
+    const directionUtils = useDirection();
+
+    const utils: StyleUtils = useMemo(
+        () => ({
+            ...theme,
+            ...directionUtils,
+            breakpoints: breakpointMediaQueries,
+            darken,
+            getValueAndUnit,
+            lighten,
+            media: mediaQueries,
+            multiply,
+            sum,
+            transparentize,
+        }),
+        [directionUtils, theme],
+    );
+
+    const nativeUtils: NativeStyleUtils = useMemo(
+        () => ({
+            ...nativeTheme,
+            ...directionUtils,
+            darken,
+            lighten,
+            transparentize,
+        }),
+        [directionUtils, nativeTheme],
+    );
+
+    const applyStyle = <TProps extends object>(
+        ...params: Record<string, unknown> extends TProps
+            ? [styleOrStyles: StyleOrStylesParam<TProps>, props?: TProps]
+            : [styleOrStyles: StyleOrStylesParam<TProps>, props: TProps]
+    ) => {
+        const [styleOrStyles, props] = params;
+
+        const invokeStyle = (style: Style<any>) => style(utils, props ?? {});
+
+        const rule = () => {
+            if (G.isArray(styleOrStyles)) {
```

Comment:
I can probably convert that, but for consistency I prefer to use type guards from ts-belt since I have this lib available. I don't want to mix it too much, because if I will use another type guard like `G.isObject` it will be strange to use `G` in one place and `Array` in another.

Reply:
makes sense. but also still a bit weird to use an external version of a built in function :(

Reply:
I see you point because this is only usage of ts-belt here. I think I can convert it to `Array.isArray` here to make you happy 😄

Reply:
I mean, I won't hold you hostage without an approval ahahah. Let's ask @tomasklim.